cloud/aws/deployer/deployer.go

```diff
@@ -14,81 +14,34 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package deployer
+package cluster
 
 import (
-	"fmt"
-	"github.com/pkg/errors"
-	"k8s.io/client-go/tools/clientcmd"
-	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
-	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"github.com/golang/glog"
 
+	clusteractuator "sigs.k8s.io/cluster-api-provider-aws/cloud/aws/actuators/cluster"
 	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/providerconfig/v1alpha1"
-	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/services/certificates"
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
 )
 
 // ProviderName is the name of the cloud provider
 const ProviderName = "aws"
 
 func init() {
-	clustercommon.RegisterClusterProvisioner(ProviderName, &AWSDeployer{})
-}
-
-// AWSDeployer implements the cluster-api Deployer interface.
-type AWSDeployer struct{}
-
-// GetIP returns the IP of a machine, but this is going away.
-func (*AWSDeployer) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
 	codec, err := v1alpha1.NewCodec()
 	if err != nil {
-		return "", errors.Wrap(err, "Failed to create codec in deployer")
+		glog.Fatalf("Could not create codec: %v", err)
 	}
 
-	status := &v1alpha1.AWSClusterProviderStatus{}
-	if err := codec.DecodeProviderStatus(cluster.Status.ProviderStatus, status); err != nil {
-		return "", errors.Wrap(err, "failed to decode cluster provider status in deployer")
-	}
-	if status.Network.APIServerELB.DNSName == "" {
-		return "", errors.New("ELB has no DNS name")
-	}
-	return status.Network.APIServerELB.DNSName, nil
-}
-
-// GetKubeConfig returns the kubeconfig after the bootstrap process is complete.
-func (*AWSDeployer) GetKubeConfig(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
-	codec, err := v1alpha1.NewCodec()
-	if err != nil {
-		return "", errors.Wrap(err, "Failed to create codec in deployer")
+	params := clusteractuator.ActuatorParams{
+		Codec: codec,
 	}
 
-	status := &v1alpha1.AWSClusterProviderStatus{}
-	if err := codec.DecodeProviderStatus(cluster.Status.ProviderStatus, status); err != nil {
-		return "", errors.Wrap(err, "failed to decode machine provider status in deployer")
-	}
-	cert, err := certificates.DecodeCertPEM(status.CACertificate)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to decode CA Cert")
-	}
-	if cert == nil {
-		return "", errors.New("certificate not found in status")
-	}
-	key, err := certificates.DecodePrivateKeyPEM(status.CAPrivateKey)
+	actuator, err := clusteractuator.NewActuator(params)
```

Comment:
This actuator is missing the aws client information, I would expect the Deployer interface functions to fail because the aws client getters are going to be nil

Reply:
NewActuator sets up the servicesgetter. Cluster controller doesn't pass them in either